api/src/main/java/com/redhat/ipaas/api/v1/rest/KubernetesClientProducer.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.ipaas.api.v1.rest;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+@ApplicationScoped
+public class KubernetesClientProducer {
+
+    @Produces
+    public KubernetesClient create() {
+        return new DefaultKubernetesClient();
```

Comment:
This needs to be addressed to the client side.

If we break the Config object, in two parts:
- properties that affect the connection _(e.g. url, keys, certificates)_
- properties that don't affect the connection _(token, credentials, timeout values)_

we could possibly allow partial client reconfiguration:

*DSL style*

     client.withRequstConfiguration(config).pods().list();

*Functional Style*

    PodList list =  client.withRequstConfiguration(config, new Function<KubernetesClient, PodList> {
            public PodList apply(KubernetesClient client) {
                  return client.pods().list();
            }
     });

*Lamda syle*

    PodList list =  client.withRequstConfiguration(config, c -> c.pods.list());


Eitherway, I think that this is outside the scope of this pull request.